### DISCUSSION TITLE
Update transitive dependencies without rebuilding of whole lock

### DISCUSSION THREAD

**Discussion by Shamilik** [renovatebot/renovate#30939] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

37-full, gitea, python, poetry

### Please tell us more about your question or problem

Hello!

`pyproject.toml` of my service looks like:
```toml
# ...
[tool.poetry.dependencies]
python = "~3.12"
my-web = "^4.7.0"
# ...
```
`setup.py` of `my-web` looks like:
```python
# ...
setup(
    name='my-web',
    version='4.7.0',
    # ...
    python_requires='>=3.10',
    install_requires=[
        'my-helpers >=1.2,<2',
        # ...
    ],
)
# ...
```

I would like to update only specified dependencies. In order to achieve this, I use config like so:
```json
{
  "//": "...",
  "packageRules": [
      {
        "matchPackagePatterns": [".*"],
        "excludePackageNames": ["my-web", "..."],
        "enabled": false
      }
  ],
  "//": "...",
}
```
It's working nice with direct dependencies.

But is there any way to update transitive one without rebuilding of whole lock with `lockFileMaintenance`? For example I made an update for `my-helpers` (dependency of `my-web`) and I want to update only it. Without Renovate it can be achieved with `poetry update --lock --no-interaction my-helpers`, but is it possible with Renovate?

**Comment by rarkins** [comment, marked as answer]
Updating transitive dependencies "specifically" like you describe is not supported
